# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/apz/src/FocusTarget.cpp
# Commit: 96da9d241051
# Full Hash: 96da9d241051d222bdd693bdb519d058a013b59c
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 8cefa694f811 -r 96da9d241051 gfx/layers/apz/src/FocusTarget.cpp
--- a/gfx/layers/apz/src/FocusTarget.cpp	Fri Mar 22 18:28:31 2019 +0000
+++ b/gfx/layers/apz/src/FocusTarget.cpp	Fri Mar 22 18:28:42 2019 +0000
@@ -221,6 +221,19 @@
   ScrollTargets target;
   target.mHorizontal = nsLayoutUtils::FindIDForScrollableFrame(horizontal);
   target.mVertical = nsLayoutUtils::FindIDForScrollableFrame(vertical);
+  if (XRE_IsContentProcess()) {
+    target.mHorizontalRenderRoot = gfxUtils::GetContentRenderRoot();
+    target.mVerticalRenderRoot = gfxUtils::GetContentRenderRoot();
+  } else {
+    target.mHorizontalRenderRoot =
+        horizontal ? gfxUtils::RecursivelyGetRenderRootForFrame(
+                         horizontal->GetScrolledFrame())
+                   : wr::RenderRoot::Default;
+    target.mVerticalRenderRoot =
+        vertical ? gfxUtils::RecursivelyGetRenderRootForFrame(
+                       vertical->GetScrolledFrame())
+                 : wr::RenderRoot::Default;
+  }
   mData = AsVariant(target);
 
   FT_LOG("Creating scroll target with seq=%" PRIu64 ", kl=%d, h=%" PRIu64